src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/filters/collection/CollectionFilter.java

```diff
@@ -0,0 +1,117 @@
+package io.stargate.sgv2.jsonapi.service.operation.model.impl.filters.collection;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.stargate.sgv2.jsonapi.api.model.command.clause.update.SetOperation;
+import io.stargate.sgv2.jsonapi.service.operation.model.impl.filters.DBFilterBase;
+import io.stargate.sgv2.jsonapi.service.shredding.model.DocValueHasher;
+import io.stargate.sgv2.jsonapi.service.shredding.model.DocumentId;
+import io.stargate.sgv2.jsonapi.util.JsonUtil;
+import java.math.BigDecimal;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public abstract class CollectionFilter extends DBFilterBase {
+  protected CollectionFilter(String path) {
+    super(path);
+  }
+
+  /**
+   * Optionally returns a <link>SetOperation</link> that should be used to update a new document
+   * created during an updset operation with the filter condition.
```

Comment:
nit: updset -> ... upsert?

Reply:
Should probably try to explain why this is Optional -- in what kinds of cases would no SetOperation be created?

Reply:
Fixed typo, I think comments is ok as-is.